src/App.tsx

```diff
@@ -17,18 +77,24 @@ export const App: React.FC = () => {
             <h1 className="title">Todos:</h1>
 
             <div className="block">
-              <TodoFilter />
+              <TodoFilter
+                handleFilter={handleFilter}
+                filters={filters}
+                handleClearQuery={handleClearQuery}
+              />
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {isLoading ? <Loader /> : <TodoList todos={filteredTodos} handleCurrentTodo={handleCurrentTodo} currentTodo={currentTodo} />}
+
             </div>
           </div>
         </div>
       </div>
 
-      <TodoModal />
+      {currentTodo !== null && (
```

Comment:
```suggestion
      {!!currentTodo && (
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**